kf_lib_data_ingest/etl/load/load.py

```diff
@@ -309,8 +309,11 @@ def _load_entity(self, entity_type, endpoint, entity_id, body, links):
             or self._get_target_id(entity_type, entity_id)
         )
 
+        # convert list/dict-like strings to their native forms and
         # don't send null attributes
-        body = {k: v for k, v in body.items() if not pandas.isnull(v)}
+        body = {
+            k: str_to_obj(v) for k, v in body.items() if not pandas.isnull(v)
```

Comment:
Hmm, I'm wondering whether this should be here or in transform stage. Idk what the best solution is but here are some thoughts ...

What if we wanted to develop an additional loader or someone wanted to do something with the transform stage's JSON output. 

In the former case, now the new loader also has to remember to do this `str_to_obj` conversion. 

In the latter case, its weird for someone to see list and dict type values in string form in a JSON file. If someone wanted to do something with this transform output, now they also have to know to `str_to_obj` on all values.

Reply:
> In the former case, now the new loader also has to remember to do this str_to_obj conversion.

I'm fine with this.

> In the latter case, its weird for someone to see list and dict type values in string form in a JSON file. If someone wanted to do something with this transform output, now they also have to know to str_to_obj on all values.

That's true. Hmm. I'll see what I can do about this.

Reply:
So here's the catch...

dicts, lists, and sets aren't hashable, and that makes Python get very angry. You can't use them as dict keys, you can't put them in sets, and pandas very much can't merge on them. So no matter what you do, if you don't convert them to strings, somewhere deep inside you're going to have to convert them to _something_ hashable (frozenset, tuple, whatever) otherwise nothing works. Now those types definitely aren't compatible with JSON, so you're going to have to use our funny jsonpickled write_json and read_json functions from common.misc.

Therefore, I think at best we could add another flag to read_json for searching for and converting object-like strings when reading the file.

Reply:
Sigh, yea let's just stick with 1